lib/route.js

```diff
@@ -572,6 +574,29 @@ function runPreParsing (err, request, reply) {
   }
 }
 
+function buildRouterOptions (options, defaultOptions) {
+  const routerOptions = options.routerOptions || {}
+  if (Object.hasOwn(options, 'ignoreTrailingSlash') || Object.hasOwn(options, 'ignoreDuplicateSlashes') || Object.hasOwn(options, 'maxParamLength') || Object.hasOwn(options, 'caseSensitive') || Object.hasOwn(options, 'allowUnsafeRegex') || Object.hasOwn(options, 'querystringParser') || Object.hasOwn(options, 'useSemicolonDelimiter') || Object.hasOwn(options, 'constraints')) {
+    FSTDEP022()
+  }
+
+  routerOptions.ignoreTrailingSlash = Object.hasOwn(routerOptions, 'ignoreTrailingSlash') ? options.routerOptions.ignoreTrailingSlash : options.ignoreTrailingSlash || defaultOptions.ignoreTrailingSlash
+  routerOptions.ignoreDuplicateSlashes = Object.hasOwn(routerOptions, 'ignoreDuplicateSlashes') ? options.routerOptions.ignoreDuplicateSlashes : options.ignoreDuplicateSlashes || defaultOptions.ignoreDuplicateSlashes
+  routerOptions.maxParamLength = Object.hasOwn(routerOptions, 'maxParamLength') ? options.routerOptions.maxParamLength : options.maxParamLength || defaultOptions.maxParamLength
+  routerOptions.allowUnsafeRegex = Object.hasOwn(routerOptions, 'allowUnsafeRegex') ? options.routerOptions.allowUnsafeRegex : options.allowUnsafeRegex || defaultOptions.allowUnsafeRegex
+
+  routerOptions.useSemicolonDelimiter = Object.hasOwn(routerOptions, 'useSemicolonDelimiter') ? options.routerOptions.useSemicolonDelimiter : options.useSemicolonDelimiter || defaultOptions.useSemicolonDelimiter
+  routerOptions.caseSensitive = Object.hasOwn(routerOptions, 'caseSensitive') ? options.routerOptions.caseSensitive : options.caseSensitive
+  routerOptions.querystringParser = Object.hasOwn(routerOptions, 'querystringParser') ? options.routerOptions.querystringParser : options.querystringParser
+  routerOptions.constraints = Object.hasOwn(routerOptions, 'constraints') ? options.routerOptions.constraints : options.constraints
```

Comment:
Does not it seem more easy to read?

```suggestion
  const propKeys = [
    'ignoreTrailingSlash',
    'ignoreDuplicateSlashes',
    'maxParamLength',
    'allowUnsafeRegex',
    'useSemicolonDelimiter',
    'caseSensitive',
    'querystringParser',
    'constraints'
  ]

  for (const key of propKeys) {
    routerOptions[key] = Object.hasOwn(routerOptions, key) ? options.routerOptions[key] : options[key] || defaultOptions[key];
  }
```

Reply:
Agree, should be updated now